Sources/NIO/Embedded.swift

```diff
@@ -630,6 +630,23 @@ public final class EmbeddedChannel: Channel {
         // This will never throw...
         try! register().wait()
     }
+    
+    /// Create a new instance.
+    ///
+    /// During creation it will automatically also register itself on the `EmbeddedEventLoop`.
+    ///
+    /// - parameters:
+    ///     - handlesr: The `ChannelHandler`s to add to the `ChannelPipeline` before register or `nil` if none should be added.
+    ///     - loop: The `EmbeddedEventLoop` to use.
+    public init(handlers: [ChannelHandler], loop: EmbeddedEventLoop = EmbeddedEventLoop()) {
+        self.embeddedEventLoop = loop
+        self._pipeline = ChannelPipeline(channel: self)
+
+        _ = try? _pipeline.addHandlers(handlers).wait()
+
+        // This will never throw...
+        try! register().wait()
+    }
```

Comment:
Can we reimplement the existing `init` in terms of this one? It'll make it easier to understand and make sure that they do the same things.

Also mind adding a unit test for this API?

Reply:
Sure to both.

I tried adding a unit test already, but found that `ChannelPipeline` doesn't expose easy functionality for getting the handlers. I can come up with something though.

Reply:
`.pipeline.handler` works fine.

Reply:
In this case the only guarantee of the API is that the handlers you pass through the initializer are present in the pipeline, so you can just try to get them out and confirm they're there.

Reply:
Surely we should guarantee the order though right?

Reply:
I have something that does check the order, it's not massively pretty

Reply:
Yeah, guaranteeing the order is a bit awkward, but the easiest way is probably to have `InboundIn == InboundOut == Array[ChannelHandler]` and then just do `fireChannelRead(Array<ChannelHandler>())`. Then, each handler can `append` itself to the data and fire it on. Should do the job.